Assets/BossRoom/Scripts/Server/Net/ServerGameNetPortal.cs

```diff
@@ -233,23 +247,75 @@ private void ApprovalCheck(byte[] connectionData, ulong clientId, MLAPI.NetworkM
-            m_ClientData[connectionPayload.clientGUID] = new PlayerData(connectionPayload.playerName, clientId);
+            //Test for Duplicate Login. 
+            if( m_ClientData.ContainsKey(connectionPayload.clientGUID))
+            {
+                if( Debug.isDebugBuild )
+                {
+                    Debug.Log($"Client GUID {connectionPayload.clientGUID} already exists. Because this is a debug build, we will still accept the connection");
+                    while( m_ClientData.ContainsKey(connectionPayload.clientGUID)) { connectionPayload.clientGUID += "_Secondary"; }
+                }
+                else
+                {
+                    ulong oldClientId = m_ClientData[connectionPayload.clientGUID].m_ClientID;
+                    StartCoroutine(CoroDisconnectClient(oldClientId, ConnectStatus.LoggedInAgain));
+                }
+            }
 
-            //TODO: GOMPS-79 handle different error cases.
+            //Test for over-capacity Login.
+            if( m_ClientData.Count >= CharSelectData.k_MaxLobbyPlayers )
+            {
+                gameReturnStatus = ConnectStatus.ServerFull;
+            }
+
+            //Populate our dictionaries with the playerData
+            if( gameReturnStatus == ConnectStatus.Success )
+            {
+                m_ClientSceneMap[clientId] = clientScene;
+                m_ClientIDToGuid[clientId] = connectionPayload.clientGUID;
+                m_ClientData[connectionPayload.clientGUID] = new PlayerData(connectionPayload.playerName, clientId);
+            }
 
             callback(false, 0, true, null, null);
 
-            //FIXME_DMW: it is weird to do this after the callback, but the custom message won't be delivered if we call it beforehand.
-            //This creates an "honor system" scenario where it is up to the client to politely leave on failure. Probably
-            //we should add a NetManager.DisconnectClient call directly below this line, when we are rejecting the connection.
-            m_Portal.S2CConnectResult(clientId, ConnectStatus.Success);
+            //TODO:MLAPI: this must be done after the callback for now. In the future we expect MLAPI to allow us to return more information as part of
+            //the approval callback, so that we can provide more context on a reject. In the meantime we must provide the extra information ourselves,
+            //and then manually close down the connection.
+            m_Portal.S2CConnectResult(clientId, gameReturnStatus);
+            if(gameReturnStatus != ConnectStatus.Success )
+            {
+                StartCoroutine(CoroDisconnectClient(clientId, gameReturnStatus));
+            }
+        }
+
+        private IEnumerator CoroDisconnectClient(ulong clientId, ConnectStatus reason)
+        {
+            const float disconnectDelay = 0.15f;
+
+            m_Portal.S2CSetDisconnectReason(clientId, reason);
+
+            yield return new WaitForSeconds(disconnectDelay);

```

Comment:
Should not wait for time, should wait for frame as a workaround
please open ticket to get this solved, so we can flush all RPCs before disconnecting.

Reply:
and add ticket id here

Reply:
Done, Issue 796 https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/issues/796